### PR TITLE
Execute powershell type provider commands in custom runspace

### PR DESCRIPTION
**Pull request by sergii-s**
Add "CustomRunspace" type which uses separate runspace for commands execution

### CODE REVIEW

**Comment by sergii-s** [comment]
@sergey-tihon hello, could you review this please ? And do you have an idea why it's failing in travis ? Thank you!

**Comment by sergey-tihon** [comment]
I am on VAC to April 27 with limited access to internet, so I will take a look as soon as I return

File: src/FSharp.Management.PowerShell/PowerShellProvider.fs [hunk 1]
```diff
@@ -52,22 +50,54 @@ type public PowerShellProvider(_cfg: TypeProviderConfig) as this =
...
-                ])
+                ]
+
+            let pty = ProvidedTypeDefinition(thisAssembly, rootNamespace, typeName, Some(baseTy))
+            pty.AddMembersDelayed(fun() -> createMembers true)
+
+            let customRunspace = ProvidedTypeDefinition("CustomRunspace", Some(typeof<obj>))
+            customRunspace.AddMember <| ProvidedConstructor([], InvokeCode = fun args -> <@@ new HostedRuntime.PSRuntimeHosted(psSnapIns.Split(';'), modules.Split(';')) :> HostedRuntime.IPSRuntime @@>)
+            customRunspace.AddInterfaceImplementation <| typeof<System.IDisposable>
+            customRunspace.AddMember <| ProvidedMethod("Dispose", [], typeof<unit>,
```
Comment (sergey-tihon) [inline src/FSharp.Management.PowerShell/PowerShellProvider.fs]: I a, really not sure that this is correct interface implementation.
Could you please test that when you do `use runspace = new PS.CustomRunspace()` is actually call `Dispose()` method?

I think that when you use `AddMember` you hide `Dispose` impl, so `(runtime :> IDisposable).Dispose()` will not call this method. I guess that we should use `customRunspace.DefineMethodOverride` to provide `Dispose` impl.

Here is a sample from `FsXaml` - https://github.com/fsprojects/FsXaml/blob/c0979473eddf424f7df83e1b9222a8ca9707c45a/src/FsXaml.Wpf.TypeProvider/XamlTypeProvider.fs#L100-L102

**Comment by sergii-s** [comment]
Thanks for the review, I'll check that

**Comment by sergii-s** [comment]
I've checked and the runspace seems to be disposed with the actual implementation, trying to put this assertion to the UT

**Comment by sergey-tihon** [comment]
@sergii-s Could you please also update docs here https://github.com/fsprojects/FSharp.Management/blob/master/docs/content/PowerShellProvider.fsx and include sample with custom runspace?

**Comment by sergey-tihon** [comment]
@sergii-s Thank you for implementation

**Comment by sergey-tihon** [comment]
Releases as [v0.4.4](https://www.nuget.org/packages/FSharp.Management/0.4.4)

**Pull request merged**